charon/src/ast/expressions_utils.rs

```diff
@@ -19,3 +20,77 @@ impl BorrowKind {
         }
     }
 }
+
+impl Place {
+    /// Compute the type of a place.
+    pub fn compute_projected_type(
+        &self,
+        type_decls: &Vector<TypeDeclId, TypeDecl>,
+        locals: &Vector<VarId, Var>,
+    ) -> Result<Ty, ()> {
```

Comment:
That makes me think we should keep the type of places in translation

Reply:
Yes, I hesitated to do that a while ago. But note that it is not a bad idea to preserve this function as a helper, or rather an equivalent function which operates on a single projection element (as the function operating on places would be useless once we add the type information).